### ISSUE TITLE
Error when key type is KeyValuePair in Dictionary

### ISSUE THREAD

**Issue by colderlord** [bridgedotnet/Bridge#3483]
In .Net i don't have an exception, but in bridge my code throws exception.

### Steps To Reproduce
[Example in Deck](https://deck.net/36f15a4d79821a4793a2204361d1d51d)
[Example in dotnetfiddle.net](https://dotnetfiddle.net/bTRzgU)

```csharp
public class Program
{
    public static void Main()
    {
        var g = new Guid("9B9AAC17-22BB-425C-AA93-9C02C5146965");
        var a = new KeyValuePair<Guid, Guid>(g, Guid.Empty);
        var dict = new Dictionary<KeyValuePair<Guid, Guid>, string>();
        dict[a] = "MyValue";
        var b = new KeyValuePair<Guid, Guid>(g, Guid.Empty);
        Console.WriteLine(dict[b]);
    }
}
```

**Comment by colderlord** [comment]
Hashes for keys does not match in dictionary. (mb only for structs?) I changed KeyValuePair by Tuple and it's works

**Comment by fabriciomurta** [comment]
Thanks for reporting the issue, also happens in Bridge 17 (https://dev.deck.net/36f15a4d79821a4793a2204361d1d51d)

**Comment by geoffreymcgill** [comment]
Pull request #3486 has been submitted with a fix for this issue. Once the PR is reviewed and any further testing is complete, I would anticipate this fix being merged into the `v17` branch, then ultimately in Bridge 17.0.

**Comment by geoffreymcgill** [comment]
@colderlord – Your original Deck is now working correctly on dev.deck. 

https://dev.deck.net/36f15a4d79821a4793a2204361d1d51d
